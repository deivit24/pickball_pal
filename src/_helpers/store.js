import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from 'redux-thunk';
import rootReducer from '../_reducers/root';

//Creating the store
const store = createStore(
  rootReducer,
  composeWithDevTools(applyMiddleware(thunk))
);

export default store;
