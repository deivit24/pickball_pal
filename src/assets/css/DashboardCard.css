#DashboardCard {
  width: 100%;
  border: none;
  box-shadow: 0 0 4px 2px #afafaf;
  border-radius: 10px;

  margin-bottom: 20px;
}

#DashboardCard .img {
  width: 100%;
  margin: 0px auto -60px auto;

  z-index: 50;
}

#DashboardCard .card-content {
  padding-top: 100px;
  background: #f4f4f4;
  z-index: -1;
}

@media only screen and (max-width: 600px) {
  #DashboardCard {
    text-align: center;
    display: flex;
    width: 100%;
    border: none;
    box-shadow: 0 0 3px 1px #afafaf;
    border-radius: 10px;
    margin-top: 0px;
    padding-top: 0px;
    margin-bottom: 10px;
  }

  #DashboardCard .img {
    width: 30%;
    margin: 0px;
    border-radius: 0%;
    z-index: 1;
  }

  #DashboardCard .card-content {
    text-align: center;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-self: center;
    padding: 0 10px;
    margin: 0 auto;
    background: #fff;
    z-index: 2;
  }
}
