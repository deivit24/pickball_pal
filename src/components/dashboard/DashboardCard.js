import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardMedia from '@material-ui/core/CardMedia';

import CardContent from '@material-ui/core/CardContent';

const useStyles = makeStyles({
  root: {
    width: '100%',
    border: 'none',
    boxShadow: '0 0 8px 2px #afafaf',
    borderRadius: '25px',
    paddingTop: '20px',
    marginBottom: '20px',
  },

  title: {
    fontSize: 14,
  },
  card: {
    paddingTop: '100px',
    background: '#f4f4f4',
    zIndex: '-1',
  },

  img: {
    width: '75%',
    margin: '0px auto -60px auto',
    borderRadius: '50%',

    zIndex: '50',
  },
  pos: {
    marginBottom: 12,
  },
});

export default function DashboardCard({ name, avatar, playing_style }) {
  const classes = useStyles();

  return (
    <>
      <Card className={classes.root} variant="outlined">
        <CardMedia
          className={classes.img}
          component="img"
          alt="Avatar Image"
          height="100%"
          image={avatar}
          title="Avatar Image"
        />
        <CardContent className={classes.card}>
          <h3 className="text-center mb-3">{name}</h3>
          <h4>My playing style:</h4>
          <p>{playing_style}</p>
        </CardContent>
      </Card>
    </>
  );
}
