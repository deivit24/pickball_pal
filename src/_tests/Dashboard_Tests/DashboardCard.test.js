import React from 'react';
import { render } from '@testing-library/react';
import DashboardCard from '../../components/dashboard/DashboardCard';
import { MemoryRouter } from 'react-router';
import { Provider } from 'react-redux';
import store from '../../_helpers/store';

//Testing DashboardCard

describe('Testing the DashboardCard', () => {
  let name = 'Test Test';
  let avatar =
    'https://icon-library.com/images/default-user-icon/default-user-icon-4.jpg';
  it('it renders DashboardCard without crashing', () => {
    render(
      <MemoryRouter>
        <Provider store={store}>
          <DashboardCard name={name} avatar={avatar} />
        </Provider>
      </MemoryRouter>
    );
  });

  it('matches snapshot', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Provider store={store}>
          <DashboardCard name={name} avatar={avatar} />
        </Provider>
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
