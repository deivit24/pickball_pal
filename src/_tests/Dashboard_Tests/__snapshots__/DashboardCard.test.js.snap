// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the DashboardCard matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiCard-root MuiPaper-outlined MuiPaper-rounded"
    id="DashboardCard"
  >
    <img
      alt="Avatar Image"
      class="MuiCardMedia-root img MuiCardMedia-media MuiCardMedia-img"
      height="100%"
      src="https://icon-library.com/images/default-user-icon/default-user-icon-4.jpg"
      title="Avatar Image"
    />
    <div
      class="MuiCardContent-root card-content"
    >
      <h3
        class="text-center mb-3"
      >
        Test Test
      </h3>
    </div>
  </div>
</DocumentFragment>
`;
